cpp/src/copying/scatter.cu

```diff
@@ -128,6 +130,8 @@ struct column_scalar_scatterer_impl {
                                      rmm::mr::device_memory_resource* mr,
                                      cudaStream_t stream) const
   {
+    CUDF_EXPECTS(source->type() == target.type(), "scalar and column types must match");
```

Comment:
Yes. This is not part of the dictionary specialization path. This line was moved to here because we need to delay the checking otherwise it would fail for the dictionary case (no scalar type matches dictionary) before even calling the specialization logic.
The test for this already exists here:
https://github.com/rapidsai/cudf/blob/fe3072f4ea0d0478705c2772935134765eab4309/cpp/tests/copying/scatter_tests.cpp#L130-L131

Reply:
Thanks for the explanation. I don't know why I couldn't find the tests.